feat: `flet doctor` CLI command

Gordon-Burns opened the pull request:
## Description

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. -->

<!-- If this PR fixes an open issue, please link to the issue here. Ex: Fixes #4562 -->

## Test Code

```python
# Test code for the review of this PR
```

## Type of change

<!-- Please delete options that are not relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I signed the CLA.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes
- [x] I have made corresponding changes to the [documentation](https://github.com/flet-dev/website) (if applicable)

## Screenshots 

<!-- Add screenshots here if applicable. -->

## Additional details

<!-- Any additional details to be known about this PR. -->

## Summary by Sourcery

New Features:
- Introduce the `flet doctor` command to diagnose potential issues with the Flet environment.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/flet-dev/flet?pullRequest=4803) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/flet-dev/flet?pullRequest=4803) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/flet-dev/flet?pullRequest=4803) it.</sub>

Gordon-Burns commented:
Tests passed and screenshot of output attached - Ready for review 
![image](https://github.com/user-attachments/assets/138e827b-0d17-4385-8790-732d3afef1c9)

ndonkoHenri commented:
Thanks for your contribution! It's awesome. 

Will have a look at it today and return to you asap.

ndonkoHenri commented:
- Running `flet doctor -v` shows the following (we have twice thesame info):
![image](https://github.com/user-attachments/assets/73e62441-38ce-47ac-93e1-7987faa7b5d2)

- Also, seems like the output is printed all at once, which takes a while. I will suggest we print info step by step like in the `flet build` command (where we make use of [rich's `status`](https://rich.readthedocs.io/en/stable/reference/status.html)) or `flutter doctor`:
![flutter-doctor-in-action](https://github.com/user-attachments/assets/6a36a9fe-8db9-47d7-9fd6-9a8ef2242451)

Gordon-Burns commented:
Thanks @ndonkoHenri  I have solved the duplicate and print issues as well as adding some additional checks in the verbose option
![image](https://github.com/user-attachments/assets/2c689056-ece5-4acd-8893-270cdbb5108c)

ndonkoHenri commented:
Great! Thanks for the changes.

Some other remarks:

- Can you please remove all the lines printing strings containing `"=="` ? And also `"Detailed check completed"` at the end. Let's please solely have the important output of the command being printed out.
- In your methods (ex: `check_flet_version`, `check_os_info`), no need to return a value as it won't be used. Then in the `handle` method we can just call the different methods, without having to store them in variables.

Gordon-Burns commented:
Picked this up over the weekend and have addressed feedback by removing the unnecessary characters. Also changed the check_vitrual_env as it did not account for someone using conda 
![image](https://github.com/user-attachments/assets/b4ebff4c-3ebd-4865-9b66-fb06e523b4db)

Gordon-Burns commented:
@ndonkoHenri  Just wanted to check in if there are anymore changes required or if this request is good ?

FeodorFitsner commented:
Re: "Flutter is not installed" on the last screenshot - in Flet 0.26.0 `flet build` is installing Flutter SDK, JDK and Android SDK if they are not installed. Other words, we are not asking/encouraging users to install those anymore. I don't know if it makes sense for `flet doctor` to check/warn on that. What do you think?

Gordon-Burns commented:
Hi @FeodorFitsner the issue asked for this to be included in the checks- but if you feel it's no longer needed I am happy to remove?

FeodorFitsner commented:
Thanks. I think it could be safely removed. Flet does no longer rely on Flutter installed by the user, but uses its private install.

Gordon-Burns commented:
Sure, I will pick this up over the weekend and push changes

Gordon-Burns commented:
@FeodorFitsner , Thanks again for the feedback this has now been addressed and pushed

ndonkoHenri commented:
Sorry @Gordon-Burns, got distracted/occupied by some other stuffs lately. 

I made some last modifications as you might have noticed: the admin privileges are non-important, and it's check has been removed. 

Your PR looks good to be merged - thanks again for picking this up!

Gordon-Burns commented:
Thanks @ndonkoHenri no worries on removing the other function

FeodorFitsner commented on sdk/python/packages/flet-cli/src/flet_cli/commands/doctor.py:
```diff
@@ -0,0 +1,80 @@
+                    "[yellow]⚠ No virtual environment or Conda detected[/yellow]"
+                )
+
+    def run_command(self, command: str) -> str:
```
Please remove this if not used.

Gordon-Burns commented:
Thanks @ndonkoHenri for removing
